client/src/components/inventory/HardwareList.tsx

```diff
@@ -17,36 +27,137 @@ export interface OwnProps {
 }
 
 interface StateProps {
-    user: User|null;
+    user: User | null;
 }
 
+type ItemsListState = {
+    isLoading: boolean;
+    activeIndices: number[];
+    searchQuery: string;
+    categories: string[];
+};
+
 type Props = StateProps & OwnProps;
 
-export class HardwareList extends React.Component<Props, { isLoading: boolean }> {
+export class HardwareList extends React.Component<Props, ItemsListState> {
 
     constructor(props: Props) {
         super(props);
         this.state = {
-            isLoading: true
+            isLoading: true,
+            activeIndices: [1],
+            searchQuery: "",
+            categories: []
         };
     }
 
+    handleClick(e: any, titleProps: any): void {
+        const {index} = titleProps;
+        const {activeIndices} = this.state;
+        const newIndex = activeIndices;
+        const currentIndexPosition = activeIndices.indexOf(index);
+        if (currentIndexPosition > -1) {
+            newIndex.splice(currentIndexPosition, 1);
+        } else {
+            newIndex.push(index);
+        }
+
+        this.setState({activeIndices: newIndex});
+    };
+
+    private containsSearchQuery(item: HwItem) {
+        const query: string = this.state.searchQuery;
+        let res = item.item_name.toLowerCase().indexOf(query) !== -1
+            || item.category.toLowerCase().indexOf(query) !== -1;
+        return res;
+    }
 
-    public render() {
 
+    public render() {
         let noRequestsMessageText = "";
         if (!this.props.requestsEnabled) {
             noRequestsMessageText = "Hardware checkout requests can't be made at this time.";
         } else if (this.props.requestsEnabled && !this.props.user) {
             noRequestsMessageText = "Sign in to request hardware.";
         }
 
-        const noRequestsMessage = !this.props.requestsEnabled || !this.props.user ? (<Message
-            title="View-only inventory"
-            warning icon>
-            <Icon name="warning sign"/>
-            {noRequestsMessageText}
-        </Message>) : "";
+        const addItemLink = (this.props.user && this.props.user.admin ?
+            <Button primary icon labelPosition='left'> <Icon
+                name='plus circle'/> Create Item </Button> : "")
```

Comment:
```suggestion
                name='plus circle'/>Create item</Button> : "")
```

don't capitalize item (in general use sentence case for buttons and most headers (exceptions are generally big headers like My Requests)